tool2schema/parameter_schema.py

```diff
@@ -36,60 +40,92 @@ def matches(parameter: Parameter) -> bool:
+        return Parameter.empty
 
-    def _add_items(self, schema: dict):
+    def _get_items(self) -> Union[str, Parameter.empty]:
         """
-        Add the items property to the given schema.
+        Get the items property to be added to the JSON schema.
+        Return `Parameter.empty` to omit the items from the schema.
         """
-        pass
+        return Parameter.empty
 
-    def _add_enum(self, schema: dict):
+    def _get_enum(self) -> Union[list[str], Parameter.empty]:
         """
-        Add the enum property to the given schema.
+        Get the enum property to be added to the JSON schema.
+        Return `Parameter.empty` to omit the enum from the schema.
         """
-        pass
+        return Parameter.empty
 
-    def _add_description(self, schema: dict):
+    def _get_description(self) -> Union[str, Parameter.empty]:
         """
-        Add the description of this parameter, extracted from the function docstring, to the given
-        schema.
+        Get the description of this parameter, extracted from the function docstring,
+        to be added to the JSON schema. Return `Parameter.empty` to omit the description
+        from the schema.
         """
         if self.docstring is None:
-            return
+            return Parameter.empty
 
         docstring = " ".join([x.strip() for x in self.docstring.replace("\n", " ").split()])
         params = re.findall(r":param ([^:]*): (.*?)(?=:param|:type|:return|:rtype|$)", docstring)
         for name, desc in params:
             if name == self.parameter.name and desc:
-                schema["description"] = desc.strip()
-                return
+                return desc.strip()
+
+        return Parameter.empty
 
-    def _add_default(self, schema: dict):
+    def _get_default(self) -> any:
         """
-        Add the default value, when present, to the given schema.
+        Get the default value for this parameter, when present, to be added to the JSON schema.
+        Return `Parameter.empty` to omit the default value from the schema.
         """
-        if self.parameter.default == Parameter.empty:
-            return
+        if self.parameter.default != Parameter.empty:
+            return self.encode_value(self.parameter.default)
 
-        schema["default"] = self.parameter.default
+        return self.parameter.default
 
     def to_json(self) -> dict:
         """
         Return the json schema for this parameter.
         """
-        json = {}
-        self._add_description(json)
-        self._add_default(json)
-        self._add_items(json)
-        self._add_type(json)
-        self._add_enum(json)
+        fields = {
+            "description": self._get_description,
+            "default": self._get_default,
+            "items": self._get_items,
+            "type": self._get_type,
+            "enum": self._get_enum,
+        }
+
+        json = {field: value for field in fields if (value := fields[field]()) != Parameter.empty}
```

Comment:
I feel like this is simpler?

```suggestion
        fields = {
            "description": self._get_description(),
            "default": self._get_default(),
            "items": self._get_items(),
            "type": self._get_type(),
            "enum": self._get_enum(),
        }

        json = {f: v for f, v in fields.items() if v != Parameter.empty}
```